app/services/proxy_status/puppetca.rb

```diff
@@ -0,0 +1,59 @@
+module ProxyStatus
+  class PuppetCA < Base
+    def certs
+      fetch_proxy_data('/certs') do
+        api.all.map do |name, properties|
+          SmartProxies::PuppetCACertificate.new([name.strip, properties['state'], properties['fingerprint'], properties["not_before"], properties["not_after"], @proxy.id])
+        end.compact
+      end
+    end
+
+    def find(name)
+      certs.find{|c| c.name == name}
+    end
+
+    def find_by_state(state)
+      certs.select{|c| c.state == state}
+    end
+
+    def expiry
+      ca_cert = find(@proxy.hostname)
```

Comment:
this will not work in all / most cases. IMHO just sort based on the oldest cert.